src/pages/ReportDetailsPage.js

```diff
@@ -56,146 +56,142 @@ const defaultProps = {
     personalDetails: {},
 };
 
-class ReportDetailsPage extends Component {
-    getPolicy() {
-        return this.props.policies[`${ONYXKEYS.COLLECTION.POLICY}${this.props.report.policyID}`];
-    }
+const ReportDetailsPage = (props) => {
+    const policy = useMemo(() => props.policies[`${ONYXKEYS.COLLECTION.POLICY}${props.report.policyID}`], [props.policies, props.report.policyID]);
+    const isPolicyAdmin = PolicyUtils.isPolicyAdmin(policy);
+    const isPolicyExpenseChat = useMemo(() => ReportUtils.isPolicyExpenseChat(props.report), [props.report]);
+    const isChatRoom = useMemo(() => ReportUtils.isChatRoom(props.report), [props.report]);
+    const isThread = useMemo(() => ReportUtils.isThread(props.report), [props.report]);
+    const chatRoomSubtitle = useMemo(() => ReportUtils.getChatRoomSubtitle(props.report), [props.report]);
 
-    getMenuItems() {
-        const menuItems = [
+    const menuItems = useMemo(() => {
+        const items = [
             {
                 key: CONST.REPORT_DETAILS_MENU_ITEM.SHARE_CODE,
                 translationKey: 'common.shareCode',
                 icon: Expensicons.QrCode,
-                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(this.props.report.reportID)),
+                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(props.report.reportID)),
             },
         ];
 
-        if (ReportUtils.isArchivedRoom(this.props.report)) {
+        if (ReportUtils.isArchivedRoom(props.report)) {
             return [];
         }
```

Comment:
Should we move this to the top of the function since it's failing fast and it doesn't depend on `const items = [` being defined before?